node/impl/client/client.go

```diff
@@ -730,9 +735,27 @@ func (a *API) clientRetrieve(ctx context.Context, order api.RetrievalOrder, ref
 			return err
 		}*/
 
+		// FIXME - this is a direct copy from https://github.com/filecoin-project/go-fil-markets/blob/v1.4.0/shared/selectors.go#L11-L16
+		// Unable to use it because we need the SelectorSpec, and markets exposes just a reified node
+		ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype.Any)
+		selspec := ssb.ExploreRecursive(
+			selector.RecursionLimitNone(),
+			ssb.ExploreAll(ssb.ExploreRecursiveEdge()),
```

Comment:
do we want a matcher here as well?

Reply:
I opted for https://github.com/filecoin-project/lotus/blob/efeaea76b97fd52203122ee33a7633da74ba9631/node/impl/client/client.go#L738-L739
Diverging didn't make sense... @hannahhoward should comment here: either we export the spec, or we adjust both